nusight2/src/client/components/theme_switcher_button/view.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useMemo, useState } from "react";
+
+import { IconButton, IconButtonProps } from "../icon_button/view";
+
+export default function ThemeSwitcherButton(props: IconButtonProps) {
+  const [isDark, setIsDark] = useState(window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches);
+  const mediaQuery = useMemo(() => window.matchMedia("(prefers-color-scheme: dark)"), []);
+
+  useEffect(() => {
+    const changeHandler = () => setIsDark(mediaQuery.matches);
+
+    mediaQuery.addEventListener("change", changeHandler);
+
+    return () => {
+      mediaQuery.removeEventListener("change", changeHandler);
+    };
+  }, [mediaQuery]);
+
+  useEffect(() => {
+    const themeColorMetaTag = document.querySelector('meta[name="color-scheme"]');
+
+    if (isDark) {
+      document.documentElement.classList.add("dark"); // Needed for styling Tailwind elements
+      if (themeColorMetaTag) {
+        themeColorMetaTag.setAttribute("content", "dark"); // Needed for default browser styling
+      }
+    } else {
+      document.documentElement.classList.remove("dark");
+      if (themeColorMetaTag) {
+        themeColorMetaTag.setAttribute("content", "light");
+      }
+    }
+  }, [isDark]);
+
+  const handleClick = () => {
+    setIsDark(!isDark);
```

Comment:
We could do that. I didn't because of NUsight being stateless. There are other things which could stay persistent (the toggles in localisation, logs, etc.). I'm not sure if this should be different.

Reply:
I'd say this one can be remembered, since it's purely visual and has no effect on functionality.  But that can be done in another PR.

Reply:
Sure. I do think it should be a separate PR though.